docs/testing/unit-tests.md

```diff
@@ -1,3 +1,247 @@
 # Unit Tests
 
-TBD
+As well fostering code quality, well-written unit tests provide a fine way for developers to quickly understand how to run a piece of code, and what it does under various conditions. It's important therefore to maintain high standards not just in our source code, but in our tests as well. 
+
+This guide is intended as a quick reference of common tools and conventions we use for Calypso unit testing. You'll be able to glean a great deal from browsing through Calypso's existing tests, but we'll try to keep this page up to date with the most recent standards.
+
+## Table of contents
+* [Running tests](running-tests)
+* [Writing tests](writing-tests)
+    * [Folder structure](#folder-structure)
+    * [Describing tests](#describing-tests)
+    * [Snapshot testing](#snapshot-testing)
+* [Examples](#examples)
+
+## Running tests
+
+See [testing-overview.md](testing-overview.md).
+
+## Writing tests
+
+We use [Jest](https://facebook.github.io/jest) testing framework for unit tests, and [Enzyme](https://github.com/airbnb/enzyme) for testing React components. You can find more information on testing components at [component tests](component-tests.md)._
+
+Though you'll still see Chai assertions and Sinon spies/mocks throughout the code base, for all new tests we use the Jest API for [globals](https://facebook.github.io/jest/docs/en/mock-function-api.html#content) (`describe`, `test`, `beforeEach` and so on) [assertions](http://facebook.github.io/jest/docs/en/expect.html#content), [mocks](http://facebook.github.io/jest/docs/en/mock-functions.html#content) and [mock function/spies](https://facebook.github.io/jest/docs/en/mock-function-api.html#content).
+
+## Folder structure
+Keep your tests in a `test` folder in your working directory. The test file should have the same name as the test subject file, and use the extension `.js`.
+
+```
++-- test
+|   +-- index.js
++-- index.jsx
+```
+
+### Describing tests
+Use 'describe' block to group test cases. Each test case should ideally describe one behaviour only.
+
+In test cases, try to describe in plain words the expected behaviour. For UI components, this might entail describing expected behaviour from a user perspective rather than explaining code internals.
+
+**Good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should disable the form submit button', () => {
+        ...
+    } );
+} );
+```
+
+**Not so good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should set this.state.disableButton to `true`', () => {
+        ...
+    } );
+} );
+```
+
+### Snapshot testing
+See [Snapshot testing](snapshot-testing.md)
+
+### Examples
+
+#### Mocking HTTP requests with Nock
+
+[Nock](https://github.com/node-nock/nock) allows us to override HTTP requests, and unit test code in isolation. For example, you might have an action  that returns a GET request in a thunk:
```

Comment:
This one is also rather considered legacy since we introduced data layer, @dmsnell can you confirm?

Reply:
I mean we can keep those examples so people could understand what's happening when such code exists, but I'm sjust saying that it might be not the best idea to use it when writing new tests.

Reply:
Ideally we wouldn't need to mock out any HTTP requests, right, unless someone is working on the `WPCOM_HTTP_REQUEST` or the HTTP pipeline. At that point, since those are core networking bits, I would hope that they know how to figure out what they are doing.

Reply:
Sounds legit. I'll remove it for now. Thanks!